### DISCUSSION TITLE
autostart

### DISCUSSION THREAD

**Discussion by kotovasia5120** [MinecraftServerControl/mscs#332] [Q&A] [answered]
Hello. How to make the server autostart when you turn on the computer?
As I understand it, I need to make a separate service with this code.
```unix
[Unit]
Description=Minecraft server

[Service]
User=minecraft
Group=minecraft
ExecStart=mscs start spigot
ExecStop=mscs stop spigot
ExecStop=sleep 10

[Install]
WantedBy=multi-user.target

``` 
Or are there other ways?

**Comment by zanix** [comment]
The [install process](https://minecraftservercontrol.github.io/docs/mscs/installation) should create the service for you.

However, if you are doing the [manual installation](https://minecraftservercontrol.github.io/docs/mscs/installation#manual-installation), part of those instructions includes adding the service file.

**Reply by kotovasia5120** [reply]
I mean the server itself, not MSCS.

**Reply by zanix** [reply]
Ah, I misunderstood.
We have [a template](https://github.com/MinecraftServerControl/mscs/blob/main/mscs%40.service) for creating a service for individual servers. I don't think the documentation has info on using this template but it's in the PR https://github.com/MinecraftServerControl/mscs/pull/297.

Maybe I should update the docs.

**Reply by zanix** [reply]
Also, you will want to disable the default service file since that would start all active servers.
```shell
sudo systemctl disable mscs.service
```

**Reply by sandain** [reply, marked as answer]
As @zanix mentions, if you are trying to autostart/autostop a single world server at a time and use systemd, you can use the service template that is already installed using our normal install method.

First, disable the default service that will start/stop all world servers:
```
sudo systemctl disable mscs
```

Next, enable the service for each of your worlds individually:
```
sudo systemctl enable mscs@world1
sudo systemctl enable mscs@world2
sudo systemctl enable mscs@spigot
```

Finally, start the service for each of your worlds individually:
```
sudo systemctl start mscs@world1
sudo systemctl start mscs@world2
sudo systemctl start mscs@spigot
```

**Reply by kotovasia5120** [reply]
Thanks, this was missing from the documentation.

**Comment by sandain** [comment]
MSCS can be auto started with a server using any number of methods.

- As @zanix mentioned, our installation documentation mentions how to automatically or manually install MSCS so that it uses either the systemd or SysV-style (e.g. upstart) methods of starting at boot.  If you are using a Linux installation that uses systemd, it would be easiest to use that method. Likewise, if you are using an older Linux install or one that avoids systemd, the old SysV-style method still works.
- However, I bet you could also get it running on boot by simply adding something like `mscs start` to your .bashrc file so that it starts when you login to the computer.
